pandas/core/computation/expressions.py

```diff
@@ -160,6 +160,8 @@ def _where_numexpr(cond, a, b):
 
 def _has_bool_dtype(x):
     try:
+        if not isinstance(x.dtype, np.dtype):
```

Comment:
huh? what is this

Reply:
@jreback this error occur dataframe have column's name 'dtype' so i temporarily change column name in this function. if this code have some issue, give me some advice. thank you

Reply:
@jreback i change the code if x is dataframe type, dont access x.dtype. i will wait your review. thank you!